Workout Core/Model/Workout/Workout.swift

```diff
@@ -519,4 +520,169 @@ public class Workout {
 		}
 	}
 	
+    private let typeRow = 0
+    private let startRow = 1
+    private let endRow = 2
+    private let durationRow = 3
+    private var distanceRow: Int? {
+        guard self.totalDistance != nil else {
+            return nil
+        }
+        
+        return 1 + durationRow
+    }
+    private var avgHeartRow: Int? {
+        guard self.avgHeart != nil else {
+            return nil
+        }
+        
+        return 1 + (distanceRow ?? durationRow)
+    }
+    private var maxHeartRow: Int? {
+        guard self.maxHeart != nil else {
+            return nil
+        }
+        
+        let base = [avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var paceRow: Int? {
+        guard self.pace != nil else {
+            return nil
+        }
+        
+        let base = [maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var speedRow: Int? {
+        guard self.speed != nil else {
+            return nil
+        }
+        
+        let base = [paceRow, maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var energyRow: Int? {
+        guard self.totalEnergy != nil else {
+            return nil
+        }
+        
+        let base = [speedRow, paceRow, maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var elevationRow: Int? {
+        let (asc, desc) = self.elevationChange
+        guard asc != nil || desc != nil else {
+            return nil
+        }
+        
+        let base = [energyRow, speedRow, paceRow, maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
```

Comment:
If I'm not mistaken these are just copied from what `WorkoutTVC` uses to configure the first section. Instead of duplicating the code why not making these public and let `WorkoutTVC` use these directly?

Reply:
You're right, but maybe I should create a new class for it like "WorkoutData", that will be used by the WorkoutTVC and Workout classes to access to theses data. What do you think?

Reply:
I'm not sure about this since `Workout` is already a wrapper for a `HKWorkout` and the model for `WorkoutTVC`.
A possible direction could be creating a sub-class for `Workout` called something like `Workout.DataIndex` (holding a weak reference to the workout in order to access what is now `self.avgHeart` and the like) exposing this properties and maybe the count of how many are non nil.
We could make the init private so only the workout itself can initialize it.